### ISSUE TITLE
DoubleAutoencoder not working

### ISSUE THREAD

**Issue by chadcwilliams** [AutoResearch/EEG-GAN#47]
The double autoencoder doesn't seem to be working. 

I did a test with two electrodes running the single autoencoder with channels_out=1 (leaving timeseries at 100 datapoints) and the double autoencoder with channels_out=1 and timeseries_out=100 (so, condensing to the same degree) and get quite different results. 

This is the single autoencoder:
![Screenshot_2023-09-08 23_04_55](https://github.com/AutoResearch/EEG-GAN/assets/26373111/e1a99e5d-6fb7-4d87-bd42-f520bf0b4d8d)

This is the double autoencoder:
![Screenshot_2023-09-08 23_38_04](https://github.com/AutoResearch/EEG-GAN/assets/26373111/c28aeeb2-ff77-434d-acfc-2f9d752e0ca8)

In the double autoencoder you can see that it is doing something, but the reconstructed data looks like the average of all the samples, rather than individual samples reconstructed. They were trained with the same number of epochs (200). With that said, the although timeseries_out in the double autoencoder equals the number of datapoints, this still adds labour to the autoencoder, so maybe it needs more epochs. I am running a very long (10000 epochs) training session to see if it does eventually work now.

I'll update if this does end up working.

**Comment by chadcwilliams** [comment]
I have confirmed it does not work. This is after 8000 epochs.
![Screenshot_2023-09-09 16_56_48](https://github.com/AutoResearch/EEG-GAN/assets/26373111/638c7dda-24b1-4c43-bf6e-10683c4c093c)

**Comment by whyhardt** [comment]
The only difference between these two was the activation function.
I've added the possibility to change that with the keyword `activation=..`
Please try `activation=sigmoid` 
That one was used in the single-AE

**Comment by chadcwilliams** [comment]
Yeah, I am just using the Double-AE in this way to see if it can work as the Single-AE does. I wouldn't do this in actual practice. Testing the activation now.

I will also do a similar test where I compare the Single-AE with target time and mimic only a time reduction in the Double-AE.

**Comment by chadcwilliams** [comment]
Daniel's last commit fixed this! Well done  @whyhardt!
